src/OpenTelemetry/Metrics/MeterProviderSdk.cs

```diff
@@ -153,12 +153,14 @@ internal sealed class MeterProviderSdk : MeterProvider
                         for (int i = 0; i < maxCountMetricsToBeCreated; i++)
                         {
                             var metricStreamConfig = metricStreamConfigs[i];
-                            var metricStreamName = metricStreamConfig?.Name ?? instrument.Name;
+                            var meterName = instrument.Meter.Name;
+                            var metricName = metricStreamConfig?.Name ?? instrument.Name;
+                            var metricStreamName = $"{meterName}.{metricName}";
 
-                            if (!MeterProviderBuilderSdk.IsValidInstrumentName(metricStreamName))
+                            if (!MeterProviderBuilderSdk.IsValidInstrumentName(metricName))
```

Comment:
Shouldn't this be `metricStreamName` still? As it is now, it again uses the instrument name, right?

Reply:
We do want to use the instrument name as the validation check is only for the instrument name. `metricStreamName` is now used to ensure the uniqueness of the combination of meter name and instrument name.

Reply:
:+1:, thanks for tackling this so quickly!